contrib_bots/lib/github_detail.py

```diff
@@ -0,0 +1,132 @@
+        config.readfp(open(self.config_path))
+        if config.get('Github', 'owner'):
+            self.owner = config.get('Github', 'owner')
+        else:
+            # Allowing undefined default repos would require multiple triage_message regexs.
+            # It's simpler to require them to be defined.
+            sys.exit('Default owner not defined')
+
+        if config.get('Github', 'repo'):
+            self.repo = config.get('Github', 'repo')
+        else:
+            sys.exit('Default repo not defined')
+
+    def usage(self):
+        # type: () -> None
+        return ("This plugin displays details on github issues and pull requests. "
+                "To refrence an issue or pr type @github-detail then anytime in the"
+                " message it's id, for example:\n"
+                "@**Github detail** #3212 zulip/#3212 zulip/zulip#3212\n"
+                "The default owner is {} and the default repo is {}.".format(self.owner, self.repo))
+
+    def triage_message(self, message, client):
+        # type: () -> bool
+        # Check the message contains a username mention, an issue idi
+        # or 'help', and that we're not replying to another bot.
+        regex = "(?:@(?:\*\*){}).+?(?:#\d+)|(?:help)".format(re.escape(client.full_name))
+        return re.search(regex, message['content']) and not message['sender_email'].endswith('-bot@zulip.com')
+
+    def format_message(self, details):
+        # type: (Dict[Text, Union[Text, int, bool]]) -> Text
+        number = details['number']
+        title = details['title']
+        link = details['html_url']
+        # Truncate if longer than 200 characters.
+        max_length = 200
+        ellipsis = '...'
+        description = ("{}{}".format(details['body'][:max_length], ellipsis)
+                       if len(details['body']) > max_length + len(ellipsis) else details['body'])
+        status = details['state'].title()
+
+        return '**[{id} | {title}]({link})** - **{status}**\n```quote\n{description}\n```'\
+            .format(id=number, title=title, link=link, status=status, description=description)
+
+    def get_details_from_github(self, owner, repo, number):
+        # type: (Text, Text, Text) -> Dict[Text, Union[Text, Int, Bool]]
+        # Gets the details of an issues or pull request
+
+        # Try to get an issue, try to get a pull if that fails
+        try:
+            r = requests.get(
+                self.GITHUB_ISSUE_URL_TEMPLATE.format(owner=owner, repo=repo, id=number))
+        except requests.exceptions.RequestException as e:
+            logging.exception(e)
+            return
+
+        if r.status_code == 404:
+            try:
+                r = requests.get(
+                    self.GITHUB_PULL_URL_TEMPLATE.format(owner=owner, repo=repo, id=number))
+            except requests.exceptions.RequestException as e:
+                logging.exception(e)
+                return
+
+        if r.status_code != requests.codes.ok:
```

Comment:
Sorry I'm not very familiar with the package, what would you change it to and why?